File: scripts/capture_design_targets.py
"""T014 · Capture design target PNGs from skin-preview.html (Python playwright)."""
from pathlib import Path
from playwright.sync_api import sync_playwright

ROOT = Path(__file__).resolve().parents[1]
SKIN = ROOT / "docs" / "design" / "skin-preview.html"
TARGETS = ROOT / "docs" / "design" / "targets"
URL_BASE = SKIN.as_uri()

SHOTS = [
    ("#layout-bazi", "bazi.png", 1120, 720),
    ("#layout-ziwei", "ziwei.png", 1120, 640),
    ("#volumes", "report-toc.png", 1120, 800),
]


def main() -> None:
    TARGETS.mkdir(parents=True, exist_ok=True)
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()
        for hash_id, filename, width, height in SHOTS:
            page.set_viewport_size({"width": width, "height": height})
            page.goto(URL_BASE + hash_id, wait_until="load")
            page.wait_for_timeout(400)
            el = page.locator(hash_id).first
            el.screenshot(path=str(TARGETS / filename))
            print("wrote", filename)
        browser.close()


if __name__ == "__main__":
    main()

File: scripts/capture-design-targets.mjs
/**
 * T014 · 从 skin-preview.html 冻结设计门禁截图
 * 用法：node scripts/capture-design-targets.mjs
 */
import { chromium } from '../frontend/node_modules/playwright/index.mjs'
import path from 'path'
import { fileURLToPath } from 'url'

const __dirname = path.dirname(fileURLToPath(import.meta.url))
const root = path.resolve(__dirname, '..')
const skinFile = path.join(root, 'docs/design/skin-preview.html')
const targetsDir = path.join(root, 'docs/design/targets')
const skinUrl = 'file:///' + skinFile.replace(/\\/g, '/')

const shots = [
  { hash: '#layout-bazi', out: 'bazi.png', width: 1120, height: 720 },
  { hash: '#layout-ziwei', out: 'ziwei.png', width: 1120, height: 640 },
  { hash: '#volumes', out: 'report-toc.png', width: 1120, height: 800 },
]

const browser = await chromium.launch()
const page = await browser.newPage()

for (const { hash, out, width, height } of shots) {
  await page.setViewportSize({ width, height })
  await page.goto(skinUrl + hash, { waitUntil: 'load' })
  await page.waitForTimeout(300)
  const el = await page.locator(hash.slice(1)).first()
  await el.screenshot({ path: path.join(targetsDir, out) })
  console.log('wrote', out)
}

await browser.close()
